Client/include/ChronicleMetadataRPCClient.h

```diff
@@ -0,0 +1,110 @@
+//
+// Created by kfeng on 12/12/22.
+//
+
+#ifndef CHRONOLOG_CHRONICLEMETADATARPCCLIENT_H
+#define CHRONOLOG_CHRONICLEMETADATARPCCLIENT_H
+
+#include <iostream>
+#include <unordered_map>
+#include <functional>
+#include <thallium.hpp>
+#include <sys/types.h>
+#include <unistd.h>
+#include "macro.h"
+#include "RPCFactory.h"
+
+class ChronicleMetadataRPCClient {
+public:
+    ChronicleMetadataRPCClient() {
+        LOGD("%s constructor is called", typeid(*this).name());
+        rpc = ChronoLog::Singleton<ChronoLogRPCFactory>::GetInstance()
+                ->GetRPC(CHRONOLOG_CONF->RPC_BASE_SERVER_PORT);
+        func_prefix = "ChronoLog";
+        switch (CHRONOLOG_CONF->RPC_IMPLEMENTATION) {
+            case CHRONOLOG_THALLIUM_SOCKETS:
+            case CHRONOLOG_THALLIUM_TCP:
+            case CHRONOLOG_THALLIUM_ROCE:
```

Comment:
Do you want to consider adding some error checks at the client side for unsupported server configurations - protocol combinations etc?